### PR TITLE
Created gradual ducking animation

### PR DESCRIPTION
**Pull request by josepharhar**
#60

### CODE REVIEW

File: src/game_state/Player.cpp [hunk 1]
```diff
@@ -4,12 +4,15 @@
...
 #include <cmath>
 #include <iostream>
+#include <algorithm>
 
 #include "TimingConstants.h"
 #include "DroppingPlatform.h"
 #include "MovingObject.h"
 
 #define MAX_DUCK_ANGLE 1.05f
+#define DUCK_FINISH_SECONDS 1.0
```
Comment (bbeck13) [inline src/game_state/Player.cpp]: Can you make this like 1/5th or 1/6th of a second this is way too slow

**Review by bbeck13 — CHANGES_REQUESTED** [review]

File: src/game_state/Player.cpp (hunk 1 above)
Comment (josepharhar) [reply src/game_state/Player.cpp]: Done, good call

File: src/game_updater/PlayerUpdater.cpp [hunk 2]
```diff
@@ -59,30 +59,35 @@ void PlayerUpdater::MovePlayer(std::shared_ptr<GameState> game_state) {
...
   } else {
-    player->SetDucking(DuckDir::NONE);
+    target_duck = Player::DuckDir::NONE;
   }
   if (ImGui::GetIO().KeysDown[GLFW_KEY_LEFT_SHIFT] ||
       ImGui::GetIO().KeysDown[GLFW_KEY_RIGHT_SHIFT]) {
-    if (player->GetDucking() == DuckDir::NONE) {
-      player->SetDucking(DuckDir::RIGHT);
+    if (player->GetDucking() == Player::DuckDir::NONE) {
+      target_duck = Player::DuckDir::RIGHT;
```
Comment (bbeck13) [inline src/game_updater/PlayerUpdater.cpp]: I'm not able to duck on this branch by pressing shift can you?

Comment (josepharhar) [reply src/game_updater/PlayerUpdater.cpp]: Fixed it, thanks for checking

**Comment by bbeck13** [comment]
CR :duck: QA :duck:

**Review by bbeck13 — APPROVED** [review]

**Pull request merged**